C-Sharp.md

```diff
@@ -1,6 +1,232 @@
+  public class Program
+  {
+    //entry point method for console applications
+   public static void Main()
+    {
+      //print lines on console
+      Console.WriteLine("Hello, World!");
+      //Reads the next line of characters from the standard input stream.Most common use is to pause program execution before clearing the console.
+      Console.ReadLine();
+    }
+  }
+}
+```
+If you are using visual Studio, follow [this](https://msdn.microsoft.com/en-us/library/k1sx6ed2.aspx) to create console application and understand your first C# program.
+
+###Types and variables
+C# is a strongly typed language. Every variable has a type.
+Every expression or statement evaluates to a value.
+There are two kinds of types in C#
+  * Value types
+  * Reference types.
+
+*_Value Types_* : Variables that are value types directly contain values. Assigning one value type variable to another copies the contained value.
+
+```chsarp
+int a = 10;
+int b = 20;
+a=b;
+Console.WriteLine(a); //prints 20
+Console.WriteLine(b); //prints 20
+```
+All these data types are also wrapped in classes in the `System namespace`. For instance
+`int` is the short name for `System.Int32` and `double` is the short form of `System.Double`.
+
+
+Few examples of built-in data types are `int, float, double, decimal, char and string`.
+
+Type | Example | Description
+--- | --- | ---
+*Integer* | `int fooInt = 7;` | **Signed 32-bit** Interger
+*Long* | `long fooLong = 3000L;` | **Signed 64-bit** integer.**L is used to specify that this variable value is of type long/ulong**
+*Double* | `double fooDouble = 20.99;` | Precision: **15-16 digits**
+*Float* | `float fooFloat = 314.5f;` | Precision: **7 digits**.**F is used to specify that this variable value is of type float**
+*Decimal* | `decimal fooDecimal = 23.3m;` | Precision: **28-29 digits**.Its more precision and smaller range makes it appropriate for **financial and monetary calculations**
+*Char* | `char fooChar = 'Z';` | A  single **16-bit Unicode character**
+*Boolean* | `bool fooBoolean = false;` | Boolean - **true & false**
+*String* |`string fooString = "\"escape\" quotes and add \n (new lines) and \t (tabs);`|**A string of Unicode characters.**
+
+For complete list of all built-in data types see [here](https://msdn.microsoft.com/en-us/library/ms228360)
+
+
+ [*_Reference types_*](https://msdn.microsoft.com/en-us/library/490f96s2.aspx) : Variables of reference types store references to their objects, which means they store the address to the location of data on the [stack](http://gribblelab.org/CBootcamp/7_Memory_Stack_vs_Heap.html#orgheadline2), also known as pointers. Actual data is stored on the [heap](http://gribblelab.org/CBootcamp/7_Memory_Stack_vs_Heap.html#orgheadline3) memory.
+ Assigning reference type to another doesn't copy the data, instead it creates the second copy of reference which points to the same location on the heap.
+
+###Flow control statements
+
```

Comment:
Which programs output you are taking about here?. Also the idea is to try there snippets in LinqPad, and see the output.
Otherwise I would prefer to use [dotnetfiddle](https://dotnetfiddle.net/2MCuIs).
But than it will be too many IDE's to choose ?